File: LeetCode-Java/Remove Duplicates from Sorted Array.java
public class Solution {
    public int removeDuplicates(int[] A) {
        if(A.length <= 0) return 0;
        int oldIndex = 1, newIndex = 1, current = A[0];
        while(oldIndex < A.length)
        {
            if(A[oldIndex] != current)
            {
                current = A[oldIndex];
                if(oldIndex > newIndex)
                {
                    int temp = A[oldIndex];
                    A[oldIndex] = A[newIndex];
                    A[newIndex] = temp;
                }
                newIndex ++;
            }
            oldIndex ++;
        }
        return newIndex;
    }
}

File: Leetcode-cpp/remove_duplicates_from_sorted_array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
       int currentIndex = 0;
       int searchIndex = 0;
       while (searchIndex < nums.size()) {
           while (searchIndex != nums.size() - 1 && nums[searchIndex] == nums[searchIndex + 1]) searchIndex++;
           nums[currentIndex++] = nums[searchIndex++];
       }
       return currentIndex;
    }
};